src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -1811,7 +1811,10 @@ bool StorageReplicatedMergeTree::tryExecuteMerge(const LogEntry & entry)
 
                 write_part_log(ExecutionStatus::fromCurrentException());
 
-                tryRemovePartImmediately(std::move(part));
+                if (storage_settings_ptr->detach_not_byte_identical_parts)
+                    forgetPartAndMoveToDetached(std::move(part), "merge-not-byte-identical");
```

Comment:
It is strange to mix `-` and `_` in part names, let's change to `_`.

Reply:
`_` is forbidden, see https://github.com/ClickHouse/ClickHouse/blob/3bd2d6b98dffd0fb3f000b2c431d09c49c0df475/src/Storages/MergeTree/IMergeTreeDataPart.cpp#L1325

Reply:
Ok.